pyarrow-stubs/_flight.pyi

```diff
@@ -196,7 +199,7 @@ class _MetadataRecordBatchReader(_Weakrefable, _ReadPandasMixin):
 class MetadataRecordBatchReader(_MetadataRecordBatchReader): ...
 
 class FlightStreamReader(MetadataRecordBatchReader):
-    def cancel(self): ...
+    def cancel(self) -> None: ...
```

Comment:
just missing a return annotation (and [no value is returned](https://github.com/apache/arrow/blob/77f70e4a122c50f28d9ff97debb8831d2558f4d3/python/pyarrow/_flight.pyx#L1065-L1068))